cg/cli/backup.py

```diff
@@ -1,32 +1,110 @@
 """Backup related CLI commands."""
 import logging
 from pathlib import Path
-from typing import Iterable, List, Optional, Union
+from typing import Iterable, List, Optional, Tuple, Union
 
 import click
 import housekeeper.store.models as hk_models
 
 from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import FileExtensions
 from cg.constants.constants import DRY_RUN, FlowCellStatus
 from cg.constants.housekeeper_tags import SequencingFileTag
+from cg.exc import FlowCellError
 from cg.meta.backup.backup import BackupAPI, SpringBackupAPI
 from cg.meta.backup.pdc import PdcAPI
-from cg.meta.encryption.encryption import EncryptionAPI, SpringEncryptionAPI
+from cg.meta.encryption.encryption import (
+    EncryptionAPI,
+    FlowCellEncryptionAPI,
+    SpringEncryptionAPI,
+)
 from cg.meta.tar.tar import TarAPI
 from cg.models.cg_config import CGConfig
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
 from cg.store import Store
 from cg.store.models import Flowcell, Sample
 
 LOG = logging.getLogger(__name__)
 
 
+def _is_encryption_possible(
+    db_flow_cell: Flowcell,
+    flow_cell: FlowCellDirectoryData,
+    complete_file_path: Path,
+    pending_file_path: Path,
+) -> Tuple[bool, Union[str, None]]:
+    """Check if requirements for encryption are meet."""
+    if db_flow_cell and db_flow_cell.has_backup:
+        return False, f"Flow cell: {flow_cell.id} is already backed-up"
+    if not flow_cell.is_flow_cell_ready():
+        return False, f"Flow cell: {flow_cell.id} is not ready"
```

Comment:
this is also used to check for when a flow cell is ready for demultiplexing, this is a bit confusing. Maybe we can add a state to the flow cell for each step in the workflow.

`is_sequenced`
`is_demultiplexed`
`is_post_processed`

then in backup use the ones that are needed. for `is_encryption_possible`. I think it can make the contract for encryption a bit more obvious from the get go.